src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs

```diff
@@ -89,20 +104,27 @@ public override PeerAddress FindPeerToConnectTo()
                 var peerExistsInConnectNode = this.NodeSettings.ConnectionManager.Connect.Any(p => p.MapToIpv6().Match(peer.NetworkAddress.Endpoint));
                 if (peerExistsInConnectNode)
                 {
+                    this.logger.LogTrace("Peer selection failed, peer exists in -connect args {0}.", peer.NetworkAddress.Endpoint);
                     peerSelectionFailed++;
                     continue;
                 }
 
                 // If the peer is already connected just continue.
                 if (this.IsPeerConnected(peer.NetworkAddress.Endpoint))
                 {
+                    this.logger.LogTrace("Peer selection failed, peer is already connected {0}.", peer.NetworkAddress.Endpoint);
                     peerSelectionFailed++;
                     continue;
                 }
 
                 break;
             }
 
+            if (peer != null)
+                this.logger.LogTrace("(-):'{0}'", peer.NetworkAddress.Endpoint);
+            else
+                this.logger.LogTrace("(-)[PEER_NULL]");
+
```

Comment:
it should be possible to just do 
`this.logger.LogTrace("(-):'{0}'", peer?.NetworkAddress.Endpoint);`

Reply:
Fixed